src/ts/content/adapters/tags/getTags.ts

```diff
@@ -1,20 +1,25 @@
 import {makeCache} from "../../../common/util/cache";
 import {TagSearchOptions, TagTrees} from "./types";
-import Sheets, {Range} from "../sheets";
+import Sheets, {Range, ValueRange} from "../sheets";
 import {parseTree} from "./parseTags";
 import {incrementColumnBy} from "../sheets/util";
 
 declare const SPREADSHEET_ID: string; // Declared in webpack DefinePlugin
 
+type TagMapper = `${string}$TAG${string}`;
+type MappedRange = {range: Range; mapper: TagMapper};
+
 const META_TAG_SHEET = "Tag Index Index";
 /**
  * The Meta Tag Sheet stores information about where tags are in the spreadsheet
  * Each row is of the form
- *   | SHEET | TOP_LEFT | WIDTH |
+ *   | SHEET | TOP_LEFT | WIDTH | MAPPER? | AS? |
  * where
  * SHEET    is the name of a sheet in the spreadsheet where tags live
  * TOP_LEFT is the top left most cell in the sheet the contains a tag
  * WIDTH    is how many columns contain tags
+ * MAPPER   is how tags should be mapped to after being consumed by BLT
```

Comment:
so is the flow of this

* add the mapper to meta sheet
* BLL returns fetches unmapped tags as tags mapped according to mapper (they're still not updated in the sheet)
* they get updated in the sheet when users save stuff (?)

Reply:
i think i'm misunderstanding, but what happens when this happens twice? could it do the mapping operation twice? ex. `"Blue author author"`

Reply:
Hmm no I'll better document then. The original sheet doesn't get changed. It's like a materialized view in databases (I think -- it has been a while). So as far as the extension is concerned, there is an extra table, but it's created out of other source data.

With only `Blue`, `Red`, and `Green`, and a mapper, our tag validator/autocomplete will see Blue`, `Red`, `Green`, Blue author`, `Red author`, and `Green author`